### DISCUSSION TITLE
Doubts about ACI outputs

### DISCUSSION THREAD

**Discussion by laralps** [QutEcoacoustics/audio-analysis#556] [Q&A]
Dear all,

Firstly thanks to the team for making Analysis Program, it is such a great tool!

I am having doubts about the output values of ACI. When I run a 96kHz underwater 10-min recording in AP (as a test), I get ACI values around 0.4-1.4. However, if I run the same audio-file in R with functions such as ACI( ) and acoustic_complexity() from 'soundecology' and 'seewave' packages, using the same frequency and 1-minute resolutions, I get completly different results in the order of ~150 per minute.

Could someone please explain to me why is that so? 

Thanks for any support,

**Comment by atruskie** [comment]
I can look into it, can you provide a sample file?

From the top of my head, we by default resample recordings to 22050 Hz. Did you disable/change that setting?


The values will never be identical - there are too many differences in low level calculations. However they should be roughly equivalent.

**Reply by laralps** [reply]
Thank you! 
A sample file can be download from [this link](https://drive.google.com/file/d/15rH-fqqepThngkt1Cpc5ZJAIFWIfktWn/view?usp=sharing). But yes, I did set the resample as 96kHz.  
Any insights will be helpful, because I'm trying to compare data obtained from both plataforms (R and AP).

**Comment by towsey** [comment]
Hi Lara, Anthony,

There are three relevant variables that I can think of:

  1.   sample rate. I have down sampled to 22kHz, 44kHz and 96kHz at various times and always got ACI values in the 0.4-1.0 range. I used 96kHz records to detect bat clicks.  I.e.  this is unlikely to be the source of your problem.
  2.  Index resolution.  I have found that for ACI, the minimum useful recording segment is about 20 seconds.  Less than that and the ACI values become noise. However, we did use 15 seconds for some 96kHz recordings to detect the bat clicks. But since you are using 60 seconds, this also is not source of the problem.
  3.  Frame size. I have only ever used 512 and 1024 samples, with 512 being the default.  These are standard for speech processing. You did not mention your frame size but I assume the default 512.

Apart from this, I don't know what to suggest other than try testing various combinations of the above.
Good luck,
Michael

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by laralps** [reply]
Thank you for the reply!
Yes, I did set the FrameLength to 512 (that is it, right?), and 50% of overlap (FrameStep: 256) both in AP and R functions.
About ACI calculation in AP, is it only calculated in the MidFrequencyBound? In the settings, is the MidFrequencyBound lower limit the LowFreqBound and the higher limit, the MidFrequencyBound itself? 
i.e.:
LowFreqBound: 250 ( = 0-250Hz), 
MidFreqBound: 1250 ( = 250 - 1250 Hz) 
HighFreqBound: 24000 ( 1250 - 24000 Hz)

I am thinking that maybe these frequency limits might be causing this difference, if ACI calculation in AP is calculated in the MidFrequencyBound... Although limiting to this frequency range still doesn't result in values on the order of ~0.6 in R functions.

**Comment by laralps** [comment]
I am trying to understand if there are any differences in the algorithm of these functions. 
However, seeing the paper of Pieretti et al. 2011 (yet for terrestrial recordings) and other papers for marine recordings (Staaterman et al. 2014; Lindseth & Lobel, 2018...), it also seems as ACI values should be higher than ~0.6-1, in the order of hundreds... 
Am I missing some basics about the mathematical calculation of the index? I apologize for my ignorance.

**Comment by towsey** [comment]
Hi Lara,
Your comment that you set frame length to 512 and 50% for overlap (ie frame step = 256) alerts me that something must be wrong. I have attached the appropriate config file, Towsey.Acoustic.yml and you will see that it states that "# WARNING: You should not change this property!!" ie SegmentOverlap, and further down at line 45, that "# Frame Overlap is NEVER used when calculating acoustic indices".

The Frame Overlap parameter is not included in the config file to prevent this from being changed. So, I think you must be using the wrong config file.

Please look at the attached config file to see if it can help you/.

Best wishes, Michael

Dr. Michael Towsey
ORCID iD http://orcid.org/0000-0002-8246-7151
QUT Ecoacoustics Research Group  |  www.Ecosounds.org<http://www.ecosounds.org/>
Science and Engineering Faculty
S block, Level 10, Room 1002, Garden Point Campus,
Queensland University of Technology,
Brisbane, Queensland 4001
Australia
Phone +61 7 3138 9381
Fax     +61 7 3138 9390
________________________________

**Reply by laralps** [reply]
Hi Towsey, 

Yes, thank you for the alert! I was using a modification of the Towsey.Acoustic.Marine.yml since my recording is also from underwater. Now I modified the  Towsey.Acoustic.yml according to my settings. 
It still results in the same range of values though (0.4-0.7). Any other idea?
